### PR TITLE
cli: flynn remote add

### LINKED ISSUE #1762: cli: Add command to (re)create the git remote for an existing app
Users should be able to just add the git remote to a clone of an application that has already been created in Flynn via a simple CLI command.

### PR DESCRIPTION
**Pull request by josephglanville**
CLI for managing git remotes for flynn applications

Closes #1762

### CODE REVIEW

File: cli/remote.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	"log"
+	"os/exec"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-docopt"
+	"github.com/flynn/flynn/controller/client"
+)
+
+func init() {
+	register("remote", runRemote, `
+usage: flynn remote <name> [-r <remote>] [-y]
```
Comment (titanous) [inline cli/remote.go]: 1. I think this should be `flynn remote add`
2. So far we always specify the app name with `flynn -a <app> remote add`, which doesn't exactly roll off the tongue and has positional issues, but we should probably support it.
3. It seems like the primary argument of this command should be the name of the remote, without having to specify `-r`.

Thoughts?

Comment (josephglanville) [inline cli/remote.go]: All of that sounds good.

Will do that and add tests. :)

File: cli/remote.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+Create a git remote that allows deploying the application via git.
+If a name for the remote is not provided 'flynn' will be used.
+
+Options:
+	-y, --yes              Skip the confirmation prompt if the git remote already exists.
+
+Examples:
+
+	$ flynn remote turkeys-stupefy-perry
+	Created remote flynn with url https://git.dev.localflynn.com/turkeys-stupefy-perry.git
```
Comment (titanous) [inline cli/remote.go]: This example looks wrong.

File: cli/remote.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		remote = "flynn"
+	}
+
+	if !inGitRepo() {
+		log.Print("Must be executed within a git repository.")
+		return nil
+	}
+
+	if remote == "" {
+		log.Print("Remote name should not be empty")
```
Comment (titanous) [inline cli/remote.go]: This doesn't look possible.

File: cli/remote.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+	for _, r := range remotes {
+		if r == remote {
+			fmt.Println("There is already a git remote called", remote)
+			if !promptYesNo("Are you sure you want to replace it?") {
+				log.Println("The remote was not created. Please, declare the desired local git remote name with --remote flag.")
+				return nil
+			}
+		}
+	}
```
Comment (titanous) [inline cli/remote.go]: Do we already have logic for this in `flynn create`? Can it be shared?

**Comment by josephglanville** [comment]
@titanous comments addressed and tests added.

File: cli/remote.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+Create a git remote that allows deploying the application via git.
+If a name for the remote is not provided 'flynn' will be used.
+
+Options:
+	-y, --yes              Skip the confirmation prompt if the git remote already exists.
+
+Examples:
+
+	$ flynn remote add turkeys-stupefy-perry
```
Comment (titanous) [inline cli/remote.go]: I think this remote name should be `staging` or similar and reflected in the line below, currently it looks like an app name (maybe move that to demonstrate the `-a` flag).

Comment (josephglanville) [inline cli/remote.go]: Done.

**Comment by titanous** [comment]
It looks like the command needs to be added to the list in `main.go`

File: cli/common.go [hunk 6]
```diff
@@ -18,3 +21,20 @@ func promptYesNo(msg string) (result bool) {
...
+func promptReplaceRemote(remote string) (bool, error) {
+	remotes, err := gitRemoteNames()
+	if err != nil {
+		return false, err
+	}
+	for _, r := range remotes {
+		if r == remote {
+			fmt.Println("There is already a git remote called", remote)
+			if !promptYesNo("Are you sure you want to replace it?") {
+				log.Println("The remote was not created. Please, declare the desired local git remote name with --remote flag.")
```
Comment (titanous) [inline cli/common.go]: Remove the comma after "Please".

**Comment by titanous** [comment]
LGTM pending my last two comments.

**Pull request merged**